### ISSUE TITLE
Product versions for binaries no longer contain commit information

### ISSUE THREAD

**Issue by elinor-fung** [dotnet/core-setup#7635]
The commit is listed as `unknown` in the product version for binaries built from core-setup.

For example, for `hostpolicy.dll` from https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.0.0-preview9-19408-09/Microsoft.NETCore.App.Runtime.win-x64.3.0.0-preview9-19408-09.nupkg, the product version is:
```
3,0,19,40809 @Commit: unknown
```

In Preview 7, the product version had the commit hash:
```
3.0.27912.14 @BuiltBy: vsagent-a00038D @Branch: internal/release/3.0 @SrcCode: https://github.com/dotnet/core-setup/tree/4da6ee6450e4dd43ef8c05925fec22d9ac3892f3. Commit Hash: 4da6ee6450e4dd43ef8c05925fec22d9ac3892f3
```

cc @dagood

**Comment by dagood** [comment]
I looked at a binlog, and it looks like this is causing it:

https://github.com/dotnet/core-setup/blob/4c96b8dd94df35b9fd73fabb374bf75c1822d691/src/corehost/build.proj#L5-L10

https://github.com/dotnet/core-setup/blob/4c96b8dd94df35b9fd73fabb374bf75c1822d691/eng/DisableSourceControlManagement.targets#L11-L19

We're relying on Arcade to generate the version part of this include file, and Arcade depends on SourceLink figuring out the repo info. I can only guess I didn't see the connection during the migration and didn't think to check. Thanks for filing this.

I'm not planning to try to fix this immediately, but I agree it should be fixed for 3.0.

**Comment by dagood** [comment]
Fixed.
